src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -78,14 +86,18 @@ public Long getDownloadSize() throws PackageDownloadException, InterruptedExcept
     }
 
     private Long getDownloadSizeWithoutRetry() throws InterruptedException, PackageDownloadException, IOException {
-        URL url = getArtifactDownloadURL(identifier, artifact.getArtifactUri().getSchemeSpecificPart());
+        String url = getArtifactDownloadURL(identifier, artifact.getArtifactUri().getSchemeSpecificPart());
 
-        HttpURLConnection httpConn = null;
-        try {
-            httpConn = connect(url);
-            int responseCode = httpConn.getResponseCode();
+        try (SdkHttpClient client = getSdkHttpClient()) {
+            HttpExecuteRequest executeRequest = HttpExecuteRequest.builder()
+                    .request(SdkHttpFullRequest.builder().uri(URI.create(url)).method(SdkHttpMethod.GET).build())
```

Comment:
We shouldn't be GETing the whole response, that will (or at least _may_) cause us to download it. We should be using HEAD

Reply:
The pre-signed URL doesn't support HEAD request yet. This improvement is added to backlog of Greengrass Component Service.

Reply:
Why is that?